archive/c/c/reverse-string.c

```diff
@@ -0,0 +1,27 @@
+#define _POSIX_C_SOURCE 200809L
+
+#include <stdio.h>
+
+int main(int argc, char **argv)
+{
+    char *line = NULL;
+    size_t bufsize;
+    ssize_t linelen;
+    size_t linepos;
+
+    /* continuously read a line from stdin */
+    while ((linelen = getline(&line, &bufsize, stdin)) != EOF) {
```

Comment:
Hey! I tested this quickly, and I found that it works great based on the brief discussion we had over at #255. However, most of the solutions in this repo use command line arguments rather than your stdin solution. 

Again, I don't mind either way as long as it's easy to execute. Some languages don't work on the command line, so I usually don't get too strict with that requirement.

Reply:
I had 2 inspirations for the user interface, the aforementioned Linux tool `rev` and the bash version of this sample program, which both use stdin. 

If you want I can change it to use command line arguments, since it's trivial either way

Reply:
Maybe command line arguments would actually be a better solution because they're more portable (`getline` is still kind of controversial even though it's been included in POSIX 09 (i think))

Reply:
I don't have much of a preference, but for the sake of consistency we should probably change the solution to leverage strings on the command line.